docs/writing-benchmarks.md

```diff
@@ -0,0 +1,78 @@
+
+#### @Setup
+
+The `@Setup` annotation is used to mark a method that sets up the necessary preconditions for your benchmark test. It serves as a preparatory step where you set up the environment for the benchmark, performing tasks such as generating data, establishing database connections, or preparing any other resources your benchmark requires. In the Kotlin/JVM and Java targets, the `@Setup` annotation can operate on two levels - `Iteration` and `Trial`. In `Iteration` level, the setup method is run before each benchmark iteration, ensuring a consistent state across all iterations. On the other hand, `Trial` level execution runs the setup method once for the entire set of benchmark method iterations, suitable when state modifications are part of the benchmark itself. Specify the level using `@Setup(Level.Trial)` or `@Setup(Level.Iteration)`; if not defined, it defaults to `Level.Trial`. The key point to remember is that the `@Setup` method's execution time is not included in the final benchmark results - the timer starts only when the `@Benchmark` method begins. This makes `@Setup` an ideal place for initialization tasks that should not impact the timing results of your benchmark. By using the `@Setup` annotation, you ensure consistency across all executions of your benchmark, providing accurate and reliable results. In the provided example, the `@Setup` annotation is used to prepare a list containing integers from 0 to a specified parameter value.
+
+
+#### @Benchmark
+
+The `@Benchmark` annotation is used to specify the methods that you want to measure the performance of. Basically, it's the actual test you're running. The code you want to benchmark goes inside this method. In our example, the `benchmarkMethod` function is annotated with `@Benchmark`, which means the toolkit will measure the performance of the operation of summing all the integers in the list.
+
+#### @TearDown
+
+The `@TearDown` annotation is used to denote a method that's executed after the benchmarking method(s), typically responsible for cleaning up or deallocating any resources or conditions that were initialized or consumed during the benchmark. For instance, if your benchmark creates temporary files or opens network connections, the method marked with `@TearDown` is where you would put the code to delete those files or close those connections. The `@TearDown` annotation helps you avoid performance bias and ensure the proper maintenance of resources and the preparation of a clean environment for the next run. In our example, the `cleanup` function annotated with `@TearDown` is used to clear the list of integers after each benchmark iteration, ensuring that each test starts with the same initial state.
```

Comment:
As you have already described in `Setup`, creating temporary files and opening network connections should be conducted there, not in a benchmark method.

Reply:
Noted!